l1-contracts/src/UniFiAVSManager.sol

```diff
@@ -212,6 +232,169 @@ contract UniFiAVSManager is UniFiAVSManagerStorage, IUniFiAVSManager, UUPSUpgrad
         $.operators[msg.sender].validatorCount -= uint128(validatorCount);
     }
 
+    /**
+     * @inheritdoc IUniFiAVSManager
+     * @dev Restricted in this context is like `whenNotPaused` modifier from Pausable.sol
+     */
+    function registerValidatorsOptimistically(ValidatorRegistrationParams[] calldata validators)
```

Comment:
In registerValidators, at the end of the function there is `operator.startDeregisterOperatorBlock = 0` which cancels operator deregistration.
This function doesn't do that. So it is possible to start deregistration, and then registerValidatorsOptimistically.

You should either cancel deregistration here as well, or prevent new validator registrations if the operator is in deregistering process.